whats_fresh/whats_fresh_api/tests/views/entry/test_inline_product.py

```diff
@@ -0,0 +1,125 @@
+
+    def test_not_logged_in(self):
+        self.client.logout()
+
+        response = self.client.get(
+            reverse('edit-vendor', kwargs={'id': '1'}))
+        self.assertRedirects(response, '/login?next=/entry/vendors/1')
+
+    def test_url_endpoint(self):
+        url = reverse('product_ajax')
+        self.assertEqual(url, '/entry/vendors/new/products/new')
+
+    def test_form_fields(self):
+        """
+        Tests to see if the form contains all of the right fields
+        """
+        response = self.client.get(reverse('product_ajax'))
+
+        fields = {'name': 'input', 'variety': 'input', 'story': 'select',
+                  'alt_name': 'input', 'description': 'textarea',
+                  'origin': 'input', 'season': 'input', 'available': 'select',
+                  'market_price': 'input', 'link': 'input',
+                  'image': 'select'}
+        form = response.context['product_form']
+
+        for field in fields:
+            # for the Edit tests, you should be able to access
+            # form[field].value
+            self.assertIn(fields[field], str(form[field]))
+
+    def test_successful_product_creation(self):
+        """
+        POST a proper "new product" command to the server, and see if the
+        new product appears in the database
+        """
+        # Create objects that we'll be setting as the foreign objects for
+        # our test product
+
+        # It needs a story, and we'll want multiple product_preparations to
+        # allow us to test the multi-product logic.
+
+        # We can't predict what the ID of the new product will be, so we can
+        # delete all of the vendors, and then choose the only vendor left
+        # after creation.
+        Product.objects.all().delete()
+
+        Story.objects.create(id=1)
+        Preparation.objects.create(id=1)
+        Preparation.objects.create(id=2)
+        Image.objects.create(id=1)
+
+        # Data that we'll post to the server to get the new Product created
+        inline_product = {'name': 'Salmon', 'variety': 'Pacific', 'story': 1,
+                          'alt_name': 'Pacific Salmon', 'origin': 'The Pacific',  # noqa
+                          'description': 'It\'s salmon -- from the Pacific!',
+                          'season': 'Always', 'available': '', 'image': 1,
+                          'market_price': '$3 a pack', 'preparation_ids': '1,2',  # noqa
+                          'link': 'https://www.youtube.com/watch?v=dQw4w9WgXcQ'}  # noqa
+
+        self.client.post(reverse('product_ajax'), inline_product)
+
+        # These values are changed by the server after being received from
+        # the client/web page.
+        inline_product['available'] = None
+        inline_product['story'] = Story.objects.get(id=inline_product['story'])
+        inline_product['image'] = Image.objects.get(id=inline_product['image'])
+
+        del inline_product['preparation_ids']
+
+        product = Product.objects.all()[0]
```

Comment:
Looks like this is where it's failing -- that Product.objects.all() has no items in it.

That means that the POST request isn't properly saving the product. To see why, you can capture the response from line 93:

``resp = self.client.post(reverse('product_ajax'), inline_product)``

and then print the content that Django is returning:

``print resp.content``

And that should help you debug the failure.